test/e2e/toolchain_test.go

```diff
@@ -63,13 +62,46 @@ func TestToolchain(t *testing.T) {
 		AssertThatSubscription(t, cheSub.Namespace, cheSub.Name, f.Client).
 			Exists().
 			HasSpec(cheSub.Spec)
+	})
+
+	t.Run("should create subscription for tekton with TektonInstallation", func(t *testing.T) {
+		// when
+		err := f.Client.Create(context.TODO(), tektonInstallation, cleanupOptions(ctx))
+
+		// then
+		require.NoError(t, err, "failed to create toolchain TektonInstallation")
+
+		err = await.WaitForTektonInstallConditions(tektonInstallation.Name, wait.UntilHasTektonStatusCondition(tekton.SubscriptionCreated(tekton.SubscriptionSuccess)))
+		require.NoError(t, err)
 
 		AssertThatSubscription(t, tektonSub.Namespace, tektonSub.Name, f.Client).
 			Exists().
 			HasSpec(tektonSub.Spec)
 	})
 
-	t.Run("should remove operatorgroup and subscription for che with installconfig deletion", func(t *testing.T) {
+	t.Run("should recreate deleted subscription for tekton", func(t *testing.T) {
+		// given
+		tektonSubscription, err := await.GetTektonSubscription()
+		require.NoError(t, err)
```

Comment:
At first sight, waiting for `GetTektonSubscription()` in e2e test looks different.

I was expecting the steps for this test case would be .. 
create_tekton_installation
wait_for_tekton_installation with ready=true
delete_tekton_subscription
wait_for_tekton_subscription to be re-created

but I see this works as we always have one and only one with same name for tekton_subscription .. but still imagine if test are running parallel and `should recreate deleted subscription for tekton` run way before `should create subscription for tekton with TektonInstallation` then is what is currently will work ??  If yes, I am fine.  If no, please do needful changes.

Reply:
@nurali-techie we'll be having only one `TektonInstallation` resource as it's cluster scoped. We don't want user to create multiple instances of `TektonInstallation`, and even anyone create it has to be with same name to trigger reconcile. 

Also I don't see how would you run it parallely using `operator-sdk` and not sure why?

Reply:
ok, make sense :+1: